### ISSUE TITLE
Old portal deprecation

### ISSUE THREAD

**Issue by mikebudzynski** [Azure/api-management-developer-portal#121]

**Comment by dcbrown16** [comment]
Is there a timeline for deprecation, i.e. months or a year after GA?

**Comment by mikebudzynski** [comment]
@dcbrown16 - not at this moment. Any concerns whether it's for example 6 months vs 1 year?

**Comment by dcbrown16** [comment]
@mikebudzynski I'm checking in for a customer in Japan through support. They don't have any particular requests regarding the period, as long as it is announced clearly. Thanks!

**Comment by dcbrown16** [comment]
We have some additional detailed questions:

1. Are there any changes in the network configuration required for API Management instances in virtual networks (internal or external) when switching from the old to new portal?
2. Do we specifically know what notifications will be provided to users when deprecation is determined/approaching? Documents/Portal notifications/emails (to whom)?

**Comment by mikebudzynski** [comment]
@dcbrown16 Thanks, good points. Just to clarify - there won't be an automatic switch to the new version. It's a manual process to configure the new portal in a similar manner to the old one, as indicated here: https://docs.microsoft.com/azure/api-management/api-management-howto-developer-portal#faq

1. Yes, VNET configuration needs to be modified to take the new portal into account. We're working on documentation on this subject.
1. API Management updates are posted here: http://aka.ms/apimupdates We'll consider sending emails as well.

**Comment by jstallm** [comment]
Once the new developer portal is GA, how do we disable the legacy developer portal explicitly if we wanted to? Say for instance that we wanted to avoid having 2 locations for documentation, and instead simply migrate exclusively to the new portal. Im wondering if this is an option.

**Comment by mikebudzynski** [comment]
This will be possible after GA - probably a few months after the release.

A workaround in the meantime would be to use the Application Gateway with API Management in VNET and not expose the old developer portal to the public.

**Comment by jstallm** [comment]
thank you @mikebudzynski , Im happy to hear this will be supported after GA. Thank you as well for the secondary idea for isolating the old developer portal from public.

**Comment by DonatasMaz** [comment]
Hello,
Currently, we are building our new API Product which will be released at the end of 2019. Here, we do have a dilemma.:
- Having in mind that MS plans to launch the GA version at the end of the year and we plan to launch our API platform at the same time, what your general advice would be for customizing/and prepare dev portal?  Should we develop our Dev Portal based on New Dev Portal or should we go with both?
Many thanks,
Donatas

**Comment by mikebudzynski** [comment]
@DonatasMaz 

> Having in mind that MS plans to launch the GA version at the end of the year and we plan to launch our API platform at the same time, what your general advice would be for customizing/and preparing dev portal? Should we develop our Dev Portal based on New Dev Portal or should we go with both?

We plan to have a production-ready version of the portal in mid-November and the new version will include significant changes to the widgets (some are already there) and default content and structure. If this timeline works for you, I suggest that you hold off until then and go with the new developer portal - especially because there won't be an automated migration between the old and the new portals.

**Comment by rholloran** [comment]
Any idea on timing of the old portal deprecation?  It would be good to know for planning purposes.  We would prefer at least 8 months notice - 12 months preferred.

**Comment by mikebudzynski** [comment]
The deprecation plans are not finalized yet. Any deprecation will be announced at least 6 month in advance

**Comment by biplab023** [comment]
@mikebudzynski - how can we  deploy the components(including images) of apim developer portal in different environments ? 
 2. how can we take the back up for the contents of the  developer portal ?
 3. I am getting Problem while signing up and signing in  from the new developer portal. a) If i try to sign up with new profiles - it always tells you are a member .b) if i try to sign in - it still shows sign in and sign up option in home screen .

**Comment by mikebudzynski** [comment]
@biplab023 

1. https://github.com/Azure/api-management-developer-portal/wiki/Migrate-portal-between-services
2. You can alter the script from the link above to perform backups.
3. Please open a separate bug report for this.

This thread is dedicated to portal deprecation, so I'll be deleting your comment in a few hours. If you'd like to report a bug, please open a new Issue. If you need assistance, please contact Azure support.

**Comment by ryangs1** [comment]
What are the migration steps to convert an existing APIM instance to use the new dev portal? Or does a new APIM instance have to be created at this time as a replacement?

**Comment by mikebudzynski** [comment]
> What are the migration steps to convert an existing APIM instance to use the new dev portal? Or does a new APIM instance have to be created at this time as a replacement?

https://docs.microsoft.com/en-us/azure/api-management/api-management-howto-developer-portal#how-can-i-migrate-from-the-old-developer-portal-to-the-new-developer-portal
No need to recreate API Management service.

**Comment by ilyabelkin** [comment]
Hi Mike, thanks for all the work on the new Developer Portal! Do you have a timeline for enabling social login providers for sign-up and sign-in? I've noticed you have a card titled "[Later] Sign in/up with online identity providers", is it planned in the short-term? We came to rely on Google for sign in since AD is not supported in some tiers of the API Management.

**Comment by mikebudzynski** [comment]
@ilyabelkin That's valuable feedback, thanks for sharing it. We don't have a timeline yet, but we'll prioritize this work item; I'll update you once we know more. Realistically, we can aim at January/February.

**Comment by ilyabelkin** [comment]
Thank you Mike, please don't deprecate the old portal before you implement this!

**Comment by mikebudzynski** [comment]
We're not deprecating the old portal yet and any deprecation will be announced more than 6 months in advance.

**Comment by seekingtheoptimal** [comment]
In the new portal, the "Try it" functionality seems more limited compared to the "legacy" portal: I couldn't find a way to test authorization for example.

Another problem I've found, related to authorization also on the legacy portal, but couldn't find any place to report it: when doing the authorization once I select my configured Oauth2 from the dropdown, in the token I get in the end, the aud (audience) value gets prepended with the domain of the AAD. My WebApi is configured with the ApplicationID as Audience, and checks are made against this one. When authorizing with the developer portal, the Audience value is https://tenant.onmicrosoft.com/ApplicationID and this causes the token validation to fail.

So, are there any plans to match the capabilities of the new portal with the old one and have ability to test authorization also?
And where can I report the issue with the authorization token audience value in the legacy portal? 
 (and if it is an issue?)

**Comment by mikebudzynski** [comment]
> In the new portal, the "Try it" functionality seems more limited compared to the "legacy" portal: I couldn't find a way to test authorization for example.

If you mean OAuth - it's a duplicate of #208 

>  And where can I report the issue with the authorization token audience value in the legacy portal?

Please open a separate issue here and we'll triage it accordingly.

**Comment by ngobw** [comment]
Will samples be still available in the new API Details page?

**Comment by azaslonov** [comment]
Hi @ngobw, representation and code samples are there already. Is there something particular missing?

**Comment by Sacratees** [comment]
@mikebudzynski, Any idea on by when we should switch to new developer portal to avoid service interruption?  
We may need at least 12 months from now to customize and roll out to new developer portal to our customers.

**Comment by mikebudzynski** [comment]
> @mikebudzynski, Any idea on by when we should switch to new developer portal to avoid service interruption?
> We may need at least 12 months from now to customize and roll out to new developer portal to our customers.

We haven't announced the legacy portal deprecation and after we announce it, you'll have at least one year to migrate to the new portal. Please, monitor this thread or aka.ms/apimupdates for updates.

However, be aware that the legacy portal won't be receiving new features. We recommend using the new portal, especially if you'd like to benefit from its extended capabilities.

**Comment by kmurari029** [comment]
Hi Mike, Hope you are doing well!
Do we have any latest update till when the Developer porta(legacy) will be supported from the mS azure side? Also, I have a simple web application hosted on legacy one currently. I want it to migrate it to new Developer portal entirely.  How to achieve this?. Do we have some configuration changes to be done for the new one?. Also since I am only the developer, and am not even knowing where the current legacy application backend/db part is managed, Please suggest how to handle this. Any documentation/support aprreciated.The application is developed in HTML, CSS, VueJS, Templates, JavaScript. Thanks in advance.

**Comment by mikebudzynski** [comment]
1. No updates on the deprecation timeline.
2. > I have a simple web application hosted on legacy one currently 

   Sorry, I don't understand that part. Your APIs are managed through API Management, so whatever they are, they will also be displayed in the new portal. You need to redo all customizations in the new portal manually. I suggest that you contact Azure support for assistance.

**Comment by kmurari029** [comment]
> 2. > I have a simple web application hosted on legacy one currently
>    
>    
>    Sorry, I don't understand this part. Your APIs are managed through API Management, so whatever they are, they will also be displayed in the new portal. You need to redo all customizations in the new portal manually. I suggest that you contact Azure support for assistance.

>>>> Thank you for your answer. Do we have any specific documentation that helps in migration of the existing legacy version application to developer potal latest version end to end?

**Comment by mikebudzynski** [comment]
> Thank you @mikebudzynski for your answer. Do we have any specific documentation that helps in migration of the existing legacy version application to developer potal latest version end to end?

You need to redo the customizations and content in the new portal - there's no workarounds. You can learn how to do it by following this tutorial: https://aka.ms/apimdocs/customizeportal
The new portal will automatically have the same users accounts, APIs, Products etc. as the legacy portal.

**Comment by fsettele** [comment]
to be honest, I like the new developer portal style, but so far it nearly covers 50% of the functionality of the current aka legacy one. And there are some bug which does not allow production use and as well there are some important features missing.

**Comment by azaslonov** [comment]
Hi @fsettele, do you mind listing all the features you specifically need from the legacy portal? We'll see what we can do about it.

**Comment by fsettele** [comment]
@azaslonov: yes sure. here are some thoughts:

- At the moment the new portal feels not very stable, so I expect, to fix all P1 and P0 bugs
- Implement issue #533 
- Better communication about which version is deployed on managed environment
- Having a Roadmap, to know which feature or fix is available
- More control on the default widgets, like in the old one, to decide which contol will be shown or what information displayed, e.g. in the search widget, I what to hide the "Show Tag" option
- The default design of the widgets shoud be improced. The legacy one was more clear. E.g. the reports widget had a taprow, and now its a loooooong list
- Allow to setup/define diffrent languages (like the parent project supports https://demo.paperbits.io/) as well with a language switcher (widget) for the customer

Happy to hear your feedback.

**Comment by mikebudzynski** [comment]
@fsettele,

> At the moment the new portal feels not very stable, so I expect, to fix all P1 and P0 bugs

All P0 bugs will soon be resolved. We're waiting on API Management service release roll outs, before we roll out fixes in the developer portal.

> Implement issue #533
> Better communication about which version is deployed on managed environment
> Allow to setup/define diffrent languages (like the parent project supports https://demo.paperbits.io/) as well with a language switcher (widget) for the customer

All are on our roadmap. The legacy developer portal didn't have those features, so we don't consider them in terms of feature parity gap.

> Having a Roadmap, to know which feature or fix is available

We use Issues and release notes in the Release section for this purpose. We did have Projects milestones, but we didn't find them informative, since we are mostly working on the reported issues. When we resume working on new features, we'll reconsider the Projects section.

> More control on the default widgets, like in the old one, to decide which contol will be shown or what information displayed, e.g. in the search widget, I what to hide the "Group by tag" option

Can you please open separate Issues for each of those feature requests? It would help us track and prioritize them.

**Comment by kwb25** [comment]
See the developer portal buttons are starting to be disabled in the APIM management UI of the Azure portal. Starting to feel like the end is near..

**Comment by fsettele** [comment]
On the "Portal Overview" page its luckily still visible. The question is how long ...

**Comment by langstonj** [comment]
Checking today on my production API Management Service I see that Developer portal (legacy) is now greyed out and non-selectable.  Developer portal is selectable.  

Does this represent the legacy portal deprecation announcement?

**Comment by mikebudzynski** [comment]
We'll be removing the legacy portal buttons from the "Overview" section of your API Management services; you can still access it from the "Portal overview" section. What you're now seeing is just an intermediary state to visually communicate the change and give you time to adjust, if you're still using the legacy portal.

> Does this represent the legacy portal deprecation announcement?

No.

**Comment by langstonj** [comment]
I guess I'm a little slow on this one but how does one enable the disabled Developer portal (legacy) option under API Management Services?

**Comment by mikebudzynski** [comment]
See my comment above. If you hover over the disabled button, you'll also see an explanation.

**Comment by langstonj** [comment]
Thanks for your patience on this….

John C Langston  |   SQL Database Administrator  |  eLynx Technologies, LLC  |  2431 E. 61st, Suite 700  |  Tulsa, OK  74136  |  www.elynxtech.com

**Comment by fsettele** [comment]
@mikebudzynski : I found an nother missing feature as our backend is secured with OAuth2. There is a restriction on the interactive Console in compartion to the legacy portal: 

> OAuth 2.0 authorization is not yet available in the interactive console of the new developer portal.

Found that on https://docs.microsoft.com/en-us/azure/api-management/api-management-howto-oauth2.

**Comment by kmurari029** [comment]
@mikebudzynski  : We have an application error on the page- "https://ericssondeveloperapi.portal.azure-api.net/products"
Its giving error on the product page- **Liquid error: Could not load file or assembly 'System.ValueTuple, Version=4.0.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040) Liquid error: Could not load file or assembly 'System.ValueTuple, Version=4.0.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)**
We haven't made any code changes to it but can see this now coming since past 2 days only. It is happening in all API'S we have this functionality. I suspect some backend/azure upgrade or config change must have happened which is showing this error. Any pointers will be appreciated. Thank you.

**Comment by mikebudzynski** [comment]
> > OAuth 2.0 authorization is not yet available in the interactive console of the new developer portal.
> Found that on https://docs.microsoft.com/en-us/azure/api-management/api-management-howto-oauth2.

OAuth 2.0 is already available in the developer portal, the documentation is outdated - we will fix it.

@kmurari029, thanks for bringing this up, we will investigate this issue.

**Comment by franktomlinson** [comment]
Can we publish the new developer portal and run it in parrallell with the legacy portal for some time?

**Comment by mikebudzynski** [comment]
> Can we publish the new developer portal and run it in parrallell with the legacy portal for some time?

Yes.

**Comment by mikebudzynski** [comment]
**Legacy developer portal is now deprecated and will retire on October 31, 2023**. See [this post in API Management's announcements repository](https://github.com/Azure/API-Management/releases/tag/deprecation-legacy-portal) for more details and migration instructions.

**Comment by kmurari029** [comment]
@mikebudzynski - I am getting this intermittent error on hit of the APIM url (https://momodeveloper.mtn.com/ )- "Something is not right. It could be a temporary glitch or a bug. Please, try again ".  The issue is not consistent, it works for some time and some time not. The project is deployed on Legacy portal and is in production. We haven't done any changes of late. However, for some other APIM it's not coming this issue. What could be the issue? Thanks in advance.

**Comment by mikebudzynski** [comment]
@kmurari029 Please, contact Azure Support - they will be able to assist you with the problem in a timely manner.

**Comment by vinoth12988** [comment]
@mikebudzynski Looks like publisher portal (Legacy) option is missing in portal overview. How do we make changes to developer portal (Legacy). We don't see any communication about this change.
![image](https://user-images.githubusercontent.com/44270477/102774555-39a02100-43b1-11eb-8078-3ad060e2d9cc.png)

**Comment by kmurari029** [comment]
> @mikebudzynski Looks like publisher portal (Legacy) option is missing in portal overview. How do we make changes to developer portal (Legacy). We don't see any communication about this change.
> ![image](https://user-images.githubusercontent.com/44270477/102774555-39a02100-43b1-11eb-8078-3ad060e2d9cc.png)

Yes, we now don't see the option- publisher portal(legacy) . How do we do edits to the custom pages created in the existing legacy portal now??

**Comment by kmurari029** [comment]
> > @mikebudzynski Looks like publisher portal (Legacy) option is missing in portal overview. How do we make changes to developer portal (Legacy). We don't see any communication about this change.
> > ![image](https://user-images.githubusercontent.com/44270477/102774555-39a02100-43b1-11eb-8078-3ad060e2d9cc.png)
> 
> @mikebudzynski Yes, we now don't see the option- publisher portal(legacy) . How do we do edits to the custom pages created in the existing legacy portal now??

@vinoth12988  I found it. Click on the Deprecated developer portal link from the Overview portal. It will open your application in admin mode. In the application top right corner, you have Administrator section. Click on it. Then click Manage option. here you get all custom pages created what we had in publisher portal(legacy)

**Comment by vinoth12988** [comment]
> > > @mikebudzynski Looks like publisher portal (Legacy) option is missing in portal overview. How do we make changes to developer portal (Legacy). We don't see any communication about this change.
> > > ![image](https://user-images.githubusercontent.com/44270477/102774555-39a02100-43b1-11eb-8078-3ad060e2d9cc.png)
> > 
> > 
> > @mikebudzynski Yes, we now don't see the option- publisher portal(legacy) . How do we do edits to the custom pages created in the existing legacy portal now??
> 
> @vinoth12988 I found it. Click on the Deprecated developer portal link from the Overview portal. It will open your application in admin mode. In the application top right corner, you have Administrator section. Click on it. Then click Manage option. here you get all custom pages created what we had in publisher portal(legacy)

@kmurari029 - Thanks!! Yes i found under Administrative interface

**Comment by vinoth12988** [comment]
Hello @kmurari029 @mikebudzynski  Any one has issue in accessing the Publisher portal link. 
Looks like publisher portal links are not opening anymore in the Administrative interface for Legacy Developer Portal?

**Comment by mikebudzynski** [comment]
@vinoth12988, we haven't made any changes in that area. Please, contact Azure support for assistance.

**Comment by verschaevesiebe** [comment]
This should be closed i.m.o, one caviat is that the "legacy developer portal" button is still shown on Azure but it doesn't do anything. Perhaps something to share with the product team directly. (show an error page)
